### PR TITLE
BB-1: Added Session related classes and implemented save logic

### PR DESCRIPTION
**Pull request by RSapeliuk**
implemented save logic, need to fully cover all with tests, also only works with the local db. Need to fix that in the future to make independent tests. Keep this in mind.

### CODE REVIEW

File: src/main/java/com/hoverla/bibernate/exception/ConnectionException.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package com.hoverla.bibernate.exception;
+
+public class ConnectionException extends RuntimeException {
```
Comment (marynamelnichuk) [inline src/main/java/com/hoverla/bibernate/exception/ConnectionException.java]: In Future let's extend all exception from BibernateApplicationException

File: pom.xml [hunk 2]
```diff
@@ -12,26 +12,54 @@
...
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter</artifactId>
+            <version>5.8.1</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.logging.log4j</groupId>
```
Comment (marynamelnichuk) [inline pom.xml]: There are duplicated depenencies (log4j), please remove them, let's decide whether log4j or slf4j we will use. Cause I used slf4j. @b-bodnar @rstkhldntsk97 @RSapeliuk @maisterkm  what do you think?

File: src/main/java/com/hoverla/bibernate/session/ResultSetParser.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import java.lang.reflect.Field;
+import java.sql.ResultSet;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class ResultSetParser {
```
Comment (marynamelnichuk) [inline src/main/java/com/hoverla/bibernate/session/ResultSetParser.java]: I have the same class in my review, so we need to decide which one we will use. I have covered my class with tests, so I propose to use my one if it's not break anything and you don't mind.

File: src/main/resources/application.properties [hunk 4]
```diff
@@ -0,0 +1,3 @@
+dataSource.user=postgres
```
Comment (marynamelnichuk) [inline src/main/resources/application.properties:1]: I belive we should not add to pull request such files, and save them in 'DO NOT COMMIT!!!!' changelist

File: src/test/java/com/customorm/factory/PersonFactory.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
+package com.customorm.factory;
+
+import com.customorm.entity.Person;
+import com.hoverla.bibernate.session.EntityKey;
+
+
+public class PersonFactory {
```
Comment (marynamelnichuk) [inline src/test/java/com/customorm/factory/PersonFactory.java]: I also have the similar class but UserFactory

**Review by marynamelnichuk — APPROVED** [review]
Let's push it and then in the separate task will merge and refactor our two PRs

File: pom.xml (hunk 2 above)
Comment (b-bodnar) [reply pom.xml]: I agree. Guys, if you don't mind, change your logging framework to slf4j

**Pull request merged**